info.limpet/src/info/limpet/UIProperty.java

```diff
@@ -0,0 +1,57 @@
+package info.limpet;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+/**
+ * Provides UI metadata for Java bean getter methods
+ */
+@Retention(RetentionPolicy.RUNTIME)
+public @interface UIProperty
+{
+	static String CATEGORY_LABEL = "Label";
+	static String CATEGORY_METADATA = "Metadata";
+	static String CATEGORY_CALCULATION = "Calculation";
+	static String CATEGORY_VALUE = "Value";
+
+	/**
+	 * @return user-friendly name of this property that will be displayed in the
+	 *         UI
+	 */
+	String name();
+
+	String category();
+
+	/**
+	 * Some properties are visible when certain condition is met.
+	 * @return a boolean expression string that must evaluate to <code>true</code> 
+	 * or <code>false</code>. The expression might refer to Java bean properties,
+	 * for example <code>"size==1"</code> is a valid expression if the bean contains
+	 * getter for a property named "size". Empty string means always visible.
+	 */
+	String visibleWhen() default "";
+	
```

Comment:
@dinko - this is a great, really powerful capability.  But, in order to justify using it, we need to make sure it gets used.  There are two ways to do this (we may need both).  The first way is to give more comment guidance, so it's a *no-brainer* to adopt using these expressions.  The other is to make sure we have lots of examples. Making lots of use of it in our code will also ensure it works as expected (via our unit tests).

Could you give another Limpet-specific example in the above comment (if possible), and also include a URL to the JECL online guidance?